docs/cloud-edge/modules/user-agent-filter.md

```diff
@@ -0,0 +1,127 @@
+# User Agent Filter
+
+---
+
+## Overview
+
+The User Agent Filter module allows you to `allow` or `deny` traffic based on the `User-Agent` header. You can define `allow` and `deny` regular expression rules that apply to the incoming `User-Agent` header.
+
+## Quickstart
+
+### Agent CLI
+
+```bash
+ngrok http 80 --ua-filter-allow="(foo/(\d)+.(\d)+)","(bar/(\d)+.(\d)+)" --ua-filter-deny="(buz/(\d)+.(\d)+)"
+```
+
+### Agent Configuration File
+
+```yaml
+tunnels:
+ example:
+   proto: https
+   addr: 80
+   ua_filter_allow: ["(foo/(\d)+.(\d)+)", "(bar/(\d)+.(\d)+)"]
+   ua_filter_deny: ["(buzz/(\d)+.(\d)+)"]
+```
+
+### Go SDK
+
+```go
+import (
+	"context"
+	"net"
+
+	"golang.ngrok.com/ngrok"
+	"golang.ngrok.com/ngrok/config"
+)
+
+func listenUserAgentFIlter(ctx context.Context) net.Listener {
+	listener, _ := ngrok.Listen(ctx,
+		config.HTTPEndpoint(
+			config.WithUserAgentFilter( []string{`(foo)/(\d)+.(\d)+`}, []string{`bar/(\d)+.(\d)+`}),
+		),
+		ngrok.WithAuthtokenFromEnv(),
+	)
+	return listener
+}
+```
+
+## Behavior
+
+### Rule Evaluation
+
+On request, the User Agent Filter module will check the incoming `User-Agent` header value against the list of defined allow and deny regular expression rules.
+
+Requests with no `User-Agent` header will be denied by deafult. If you wish to accept these you can add a allow rule for empty strings: `^$`
+
```

Comment:
```suggestion
```

Reply:
I think this makes the prettier mad

Reply:
I ran format after remove the newline and it re-adds it

Reply:
I tried to make the others a list, so was trying to remove the line between them:

- Requests with
- Requests without